assets/src/edit-story/app/media/useMediaReducer/actions.js

```diff
@@ -64,10 +64,24 @@ export const removeProcessing = (dispatch) => ({ id }) => {
   dispatch({ type: types.REMOVE_PROCESSING, payload: { id } });
 };
 
-export const updateMediaElement = (dispatch) => ({ id, posterId, poster }) => {
+export const updateMediaElement = (dispatch) => ({
+  id,
+  posterId,
+  poster,
+  posterWidth,
+  posterHeight,
+  posterGenerated,
+}) => {
   dispatch({
     type: types.UPDATE_MEDIA_ELEMENT,
-    payload: { id, posterId, poster },
+    payload: {
+      id,
+      posterId,
```

Comment:
Here and elsewhere below, it looks like what we now have is the poster resource. I think we should just allow that.

```
element = {
  type: video,
  resource: {/* video resource itself */, poster: {/*default poster resource*/}}
  poster: {/* overriden resource for the element itself */}
  ...
}
```

Otherwise soon enough all of the resource properties will leak prefixed with "posterX" everywhere.

It looks like we still need `posterGenerated` boolean here though.

Alternatively, we could try to immediately overwrite the resource's width and height with the poster's dimensions and maybe we won't need `posterWidth/Height` downstream anywhere. I made a similar comment below as well.

Reply:
Is this refactor really required so close to alpha.